gulp/tasks/preview.js

```diff
@@ -11,81 +11,80 @@
-        childProcess.execSync(
-            'jekyll build --config _config.yml,' + previewConfigFile + ' --destination ' + previewSiteDir
-        );
+gulp.task('jekyll-build-preview', function() {
+    var branch = gitUtils.currentBranch(),
+        previewBaseUrl = '/' + branch + '/';
+    // Create a temporary Jekyll configuration file which specifies the base URL for the preview site
+    childProcess.execSync(
+        'echo \'baseurl: ' + previewBaseUrl + '\' > ' + previewConfigFile
+    );
 
-        // Remove the configuration file since it is no longer needed
-        childProcess.execSync('rm ' + previewConfigFile);
-    });
+    // Generate the preview version of the site
+    console.log('Generating preview for branch ' + branch);
+    childProcess.execSync(
+        'jekyll build --config _config.yml,' + previewConfigFile + ' --destination ' + previewSiteDir
+    );
 
-    gulp.task('preview-webpack', function() {
-        var outputPath =  previewSiteDir + '/public/',
-            branch = gitUtils.currentBranch();
-        process.env.SITE_ROOT = '/' + branch + '/';
-        return gulp.src('')
-            .pipe(webpack(require('../../webpack.config.js')))
-            .pipe(gulp.dest(outputPath));
-    });
+    // Remove the configuration file since it is no longer needed
+    childProcess.execSync('rm ' + previewConfigFile);
+});
 
-    gulp.task('upload-preview', function() {
-        var branch = gitUtils.currentBranch();
-        if (previewDomain) {
-            childProcess.execSync(
-                'aws s3 sync ' + previewSiteDir + ' s3://' + previewDomain + '/' + branch
-            );
-            console.log('Preview site ready at http://' + previewDomain + '/' + branch);
-        } else {
-            console.log(
-                'No preview domain specified. Please export environment variable S3_PREVIEW_DOMAIN and try again.'
-            );
-        }
-    });
+gulp.task('preview-webpack', function() {
+    var outputPath = previewSiteDir + '/public/',
+        branch = gitUtils.currentBranch();
+    process.env.SITE_ROOT = '/' + branch + '/';
+    return gulp.src('')
+        .pipe(webpack(require('../../webpack.config.js')))
+        .pipe(gulp.dest(outputPath));
+});
 
-    gulp.task('show-preview', function() {
-        var branch = gitUtils.currentBranch();
+gulp.task('upload-preview', function() {
+    var branch = gitUtils.currentBranch();
+    if (previewDomain) {
         childProcess.execSync(
-            'open http://' + previewDomain + '/' + branch
+            'aws s3 sync ' + previewSiteDir + ' s3://' + previewDomain + '/' + branch
         );
-    });
-
-    gulp.task('remove-preview', function() {
-        var branch = gitUtils.currentBranch();
-        childProcess.execSync(
-            'aws s3 rm --recursive  s3://' + previewDomain + '/' + branch
+        console.log('Preview site ready at http://' + previewDomain + '/' + branch);
+    } else {
+        console.log(
```

Comment:
Nit: I'm not entirely sure if the output will change, but I think this could be [`console.error()`](https://nodejs.org/api/console.html#console_console_error_data) instead.  Maybe it changes something color on some consoles or error capturing.